### PR TITLE
Fix Verbosity Bug

### PR DESCRIPTION
**Pull request by unparalleled-js**
Hello, I'm Jules.

This PR is rather simple, it does two things:

1.) Git ignores .idea (from using Pycharm)
2.) Fixes verbosity issue where the margin was uncalcable because it's variables were requiring non-verbose mode unnecessarily, causing DivisionByZeroError

** Side Note ** 

However, I have been working on a python CB-PRO bot (which I like to call CBAT) for a little while now... but it is nowhere near as mature as yours, in a mathematical sense. I am in the process of experimenting with `pycryptobot` now.  What I'd like to do in the long run is add CBAT's CLI interface on top of the `pycryptobot` project.  CBAT uses the `click` framework and `setup.py` to make it more like a python package that is easy to add more CLI commands to.. Basically, I find it helpful to have other Coinbase Pro commands, such as listing open orders, placing orders etc, as one-off commands as well as the functionality of the bot, if that makes sense.

I'd like to know your thoughts on this, if it would be worth my time to make it a PR worthy interface for this project. I think it would be really neat!

Another thing it would add is the use of keyring, which is more secure than the config file. We can make a backwards compat way to handle the config.

As a closing thought, it seems that, historically, trading bots have been a private affair.  I applaud your work on this project and am very thankful to have come across it at the time that I did. Cheers!

### CODE REVIEW

**Comment by unparalleled-js** [comment]
Another simple thing I'd like to add, that CBAT has, is the use of keyring - it is a more secure option that a json config because less chance of accidents.

File: pycryptobot.py [hunk 1]
```diff
@@ -737,7 +737,8 @@ def executeJob(sc, market, granularity, tradingData=pd.DataFrame()):
             print ('   Buy Count :', buy_count)
             print ('  Sell Count :', sell_count, "\n")
 
-            print ('      Margin :', str(truncate((((sell_sum - buy_sum) / sell_sum) * 100), 2)) + '%', "\n")
+            margin_decimal = (sell_sum - buy_sum) / sell_sum if sell_sum else 0
+            print('      Margin :', str(truncate((margin_decimal * 100), 2)) + '%', "\n")
```
Comment (unparalleled-js) [inline pycryptobot.py:741]: `python3 pycryptobot.py --market ETH-USD --granularity 3600 --sim fast --verbose 1`
occasionally causes `DivisionByZeroError`

Comment (unparalleled-js) [reply pycryptobot.py:741]: Unless this is the cause of another bug? I was just assuming there were not sells or something

Comment (unparalleled-js) [reply pycryptobot.py:741]: Looking at this closer, it seems to be related to the verbosity...

File: pycryptobot.py [hunk 2]
```diff
@@ -689,27 +689,28 @@ def executeJob(sc, market, granularity, tradingData=pd.DataFrame()):
...
-                
-                    buy_sum = buy_sum + last_buy_price_minus_fees
-                    sell_sum = sell_sum + sell_price
-                else:
+                if is_verbose == 1:
                     print('--------------------------------------------------------------------------------')
                     print('|                      *** Executing TEST Sell Order ***                        |')
                     print('--------------------------------------------------------------------------------')
+
+                sell_price = float(str(truncate(price, precision)))
```
Comment (unparalleled-js) [inline pycryptobot.py:697]: I believe we still want to see the Margin when using verbose mode during non-live mode

File: pycryptobot.py [hunk 3]
```diff
@@ -737,7 +738,8 @@ def executeJob(sc, market, granularity, tradingData=pd.DataFrame()):
             print ('   Buy Count :', buy_count)
             print ('  Sell Count :', sell_count, "\n")
 
-            print ('      Margin :', str(truncate((((sell_sum - buy_sum) / sell_sum) * 100), 2)) + '%', "\n")
+            margin_decimal = (sell_sum - buy_sum) / sell_sum if sell_sum else 0
```
Comment (unparalleled-js) [inline pycryptobot.py:741]: This just prevents potential DivisionByZeroError, which is what was happening.  I believe there is still the possibility that this number could be zero, or else I'd opt for leaving it the way it is for the sake of catching potential issues.

File: pycryptobot.py (hunk 2 above)
Comment (whittlem) [reply pycryptobot.py:697]: Thanks, nice addition.

File: pycryptobot.py (hunk 3 above)
Comment (whittlem) [reply pycryptobot.py:741]: Thanks, I think I fixed this already in the new major version I've been working on. The new version supports both Binance and Coinbase Pro. If you check out the "binance" branch you can see the new code. I am pretty sure I have a good fix for this so I will check how I did it and apply the update.

Comment (whittlem) [reply pycryptobot.py:741]: This is how I did it in the "binance" version which resolved that DivisionByZeroError.

                if sell_count > 0:
                    print ('      Margin :', str(app.truncate((((sell_sum - buy_sum) / sell_sum) * 100), 2)) + '%', "\n")
                    
I'll use your update for this version. Thanks.

**Review by whittlem — APPROVED** [review]
Thank you for the updates.

**Pull request merged**